tests/cpp/test_multidevice_matmul.cpp

```diff
@@ -0,0 +1,302 @@
+// clang-format off
+/*
+ * SPDX-FileCopyrightText: Copyright (c) 2023-present NVIDIA CORPORATION & AFFILIATES.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+// clang-format on
+#include <gtest/gtest.h>
+
+#include <codegen.h>
+#include <device_lower/lower2device.h>
+#include <executor.h>
+#include <expr_evaluator.h>
+#include <fusion.h>
+#include <fusion_segmenter.h>
+#include <ir/all_nodes.h>
+#include <ir/interface_nodes.h>
+#include <ir/iostream.h>
+#include <ir/printer.h>
+#include <ir/utils.h>
+#include <iter_visitor.h>
+#include <kernel_cache.h>
+#include <kernel_ir.h>
+#include <mma_type.h>
+#include <ops/all_ops.h>
+#include <preseg_passes/allocation_order_inference.h>
+#include <preseg_passes/optimization_pass.h>
+#include <scheduler/mma_utils.h>
+#include <scheduler/utils.h>
+#include <tests/cpp/multidevice.h>
+#include <tests/cpp/validator.h>
+#include <transform_replay.h>
+#include <transform_rfactor.h>
+
+namespace nvfuser {
+
+class DistributedMatmulTest : public MultiDeviceTest {
+ protected:
+  DistributedMatmulTest() : optimization_guard_(false) {
+    DisableOptionsGuard::getCurOptions().set(DisableOption::MatmulExprEval);
+  }
+
+  void SetUp() {
+    if (!deviceMajorMinorCheck(8)) {
+      GTEST_SKIP() << "Distributed matmul tests require Ampere or newer";
+    }
+    MultiDeviceTest::SetUp();
+    num_devices = communicator->size();
+  }
+
+  MultiDeviceExecutorParams executor_params{
+      .use_fusion_executor_cache = true,
+      .skip_auto_scheduling = false,
+      .cache_fusion_executor = false};
+  int num_devices;
```

Comment:
```suggestion
  int num_devices_;
```

Also, can this and `executor_params` be moved to private?